src/components/CardEditor/CardEditForm/CardEditFormContent.jsx

```diff
@@ -0,0 +1,198 @@
+    chartType_BAR: 'Bar',
+    chartType_LINE: 'Line',
+    barChartType_SIMPLE: 'Simple',
+    barChartType_GROUPED: 'Grouped',
+    barChartType_STACKED: 'Stacked',
+    barChartLayout_HORIZONTAL: 'Horizontal',
+    barChartLayout_VERTICAL: 'Vertical',
+    cardTitle: 'Card title',
+    description: 'Description (Optional)',
+    size: 'Size',
+    selectASize: 'Select a size',
+    timeRange: 'Time range',
+    selectATimeRange: 'Select a time range',
+  },
+  getValidDataItems: null,
+  getValidTimeRanges: null,
+  dataItems: [],
+};
+
+export const defaultTimeRangeOptions = {
+  last24Hours: 'Last 24 hrs',
+  last7Days: 'Last 7 days',
+  lastMonth: 'Last month',
+  lastQuarter: 'Last quarter',
+  lastYear: 'Last year',
+  thisWeek: 'This week',
+  thisMonth: 'This month',
+  thisQuarter: 'This quarter',
+  thisYear: 'This year',
+};
+
+/**
+ * Returns card size and dimensions labels
+ * @param {string} size
+ * @param {Object<string>} i18n
+ * @returns {string}
+ */
+export const getCardSizeText = (size, i18n) => {
+  const sizeName = i18n[`cardSize_${size}`];
+  const sizeDimensions = `(${CARD_DIMENSIONS[size].lg.w}x${CARD_DIMENSIONS[size].lg.h})`;
+  return `${sizeName} ${sizeDimensions}`;
+};
+
+const CardEditFormContent = ({
+  cardJson,
+  onChange,
+  i18n,
+  dataItems,
+  getValidDataItems,
+  getValidTimeRanges,
+}) => {
+  const { title, description, size, type } = cardJson;
+  const mergedI18n = { ...defaultProps.i18n, ...i18n };
+
+  const baseClassName = `${iotPrefix}--card-edit-form`;
+
+  return (
+    <>
+      <div className={`${baseClassName}--input`}>
+        <TextInput
+          id="title"
+          labelText={mergedI18n.cardTitle}
+          light
+          onChange={(evt) => onChange({ ...cardJson, title: evt.target.value })}
+          value={title}
+        />
+      </div>
+      <div className={`${baseClassName}--input`}>
+        <TextArea
+          id="description"
+          labelText={mergedI18n.description}
+          light
+          onChange={(evt) =>
+            onChange({ ...cardJson, description: evt.target.value })
+          }
+          value={description}
+        />
+      </div>
+      <div className={`${baseClassName}--input`}>
+        <Dropdown
+          id="size"
+          label={mergedI18n.selectASize}
+          direction="bottom"
+          itemToString={(item) => item.text}
+          items={(
+            ALLOWED_CARD_SIZES_PER_TYPE[type] ?? Object.keys(CARD_SIZES)
+          ).map((cardSize) => {
+            return {
+              id: cardSize,
+              text: getCardSizeText(cardSize, mergedI18n),
+            };
+          })}
+          light
+          selectedItem={{ id: size, text: getCardSizeText(size, mergedI18n) }}
+          onChange={({ selectedItem }) => {
+            onChange({ ...cardJson, size: selectedItem.id });
+          }}
+          titleText={mergedI18n.size}
+        />
+      </div>
+      {type === CARD_TYPES.TIMESERIES && (
```

Comment:
This is okay for now, but we'll want to extract any component-specific items out in the future to keep this component cleaner